Updated MCP23017 Component To Cover 8 & 16 Port Chips

TomFahey opened the pull request:
## Description:

Updated documentation to cover both MCP23017 and MCP23008 IO expanders.

**Related issue (if applicable):** fixes <https://github.com/esphome/feature-requests/issues/41>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#649

## Checklist:

  - [y] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

TomFahey committed:
Updated MCP23017 Component To Cover 8 & 16 Port Chips

OttoWinter commented on components/mcp230xx.rst:
```diff
@@ -0,0 +1,75 @@
+MCP230xx I/O Expander
```
Please also remove the MCP23017 .rst file then.

TomFahey replied:
Done.

OttoWinter commented on components/mcp230xx.rst:
```diff
@@ -0,0 +1,75 @@
+.. seo::
+    :description: Instructions for setting up MCP23017 or MCP23008 digital port expanders in ESPHome.
+    :image: mcp230xx.png
+
```
Add a short paragraph at the top about how this page is for both MCP23017 and MCP23008.

Then make MCP23017 and 008 into their own "sections" (--- underline)

TomFahey replied:
Done.

OttoWinter commented on images/mcp230xx.svg:
```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
+<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="301px" height="61px" viewBox="0 0 301 61" enable-background="new 0 0 301 61" xml:space="preserve">  <image id="image0" width="301" height="61" x="0" y="0"
+    xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAS0AAAA9CAQAAABSS/dKAAAABGdBTUEAALGPC/xhBQAAACBjSFJN
+AAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAAmJLR0QA/4ePzL8AAAAHdElN
```
This is not an SVG - it is a PNG embedded in a SVG.

I can make a MCP230XX svg banner later if you'd like.

TomFahey replied:
That'd be ideal, thanks!

OttoWinter commented:
Also: You're creating this PR against the `current` branch - new features should be created against the `next` branch in the documentation.

TomFahey committed:
Made requested changes, adding short paragraph at top of MCP230xx.rst
explaining the page is for MCP23008 and MCP23017, also deleted now
obsolete MCP23017.rst

TomFahey committed:
Added separate sections for MCP23017, MCP23008

TomFahey commented:
Hi, sorry about that - I ended up being quite busy around the time I submitted this, then completely forgot about it! This is actually my very first pull request and "contribution" (if it can be called that...) to open source software, so sorry if I seem a bit clueless!

glmnet commented:
Welcome to otto world. You’re going to learn a lot here.

El El jue, 11 jul. 2019 a las 16:16, TomFahey <notifications@github.com>
escribió:

> Hi, sorry about that - I ended up being quite busy around the time I
> submitted this, then completely forgot about it! This is actually my very
> first pull request and "contribution" (if it can be called that...) to open
> source software, so sorry if I seem a bit clueless!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/esphome/esphome-docs/pull/277?email_source=notifications&email_token=AAAXPITEVFV72RJKM7QCN5TP66BKDA5CNFSM4HZN7FOKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZXWU6Q#issuecomment-510618234>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAXPIS4TDTTOT6JNGZGZILP66BKDANCNFSM4HZN7FOA>
> .
>
-- 
*GLM*

TomFahey committed:
Merge branch 'next' into update-mcp23017-mcp230xx

OttoWinter committed:
Remove MCP23017 page (merged with new one)
